### PR TITLE
reduce sources passed to the compiler to the dependent ones

### PR DESCRIPTION
**Pull request by moosq**

### CODE REVIEW

**Review by 5nord — CHANGES_REQUESTED** [review]

File: k3/k3.go [hunk 1]
```diff
@@ -204,6 +206,74 @@ func NewT3XF(vars map[string]string, t3xf string, srcs ...string) []*proc.Cmd {
...
+func CollectImports(name string) []string {
+	tree := ttcn3.ParseFile(name)
+	if tree == nil || tree.Root == nil {
+		return nil
+	}
+
+	ret := make([]string, 0, 10)
+	ast.Inspect(tree.Root, func(n ast.Node) bool {
+		switch n := n.(type) {
+		case *ast.NodeList, *ast.Module, *ast.ModuleDef:
```
[the remark this answers is not in the document]
Comment (moosq) [reply k3/k3.go:230]: is this really necessary? I thought import statements are not allowed inside groups

[the remark this answers is not in the document]
Comment (5nord) [reply k3/k3.go:230]: I am not aware of any restrictions.

Instead of walking the AST manually you could also use the ttcn3.Tree.Imports method.

**Comment by 5nord** [comment]
I provided an alternative: https://github.com/nokia/ntt/pull/554
Maybe this would be sufficient?

**Comment by moosq** [comment]
has been resolved by #554

**Pull request closed without merging**